tardis/plasma/properties/nlte_rate_equation_solver.py

```diff
@@ -0,0 +1,367 @@
+import pandas as pd
+import numpy as np
+
+from tardis.plasma.properties.base import ProcessingPlasmaProperty
+
+__all__ = [
+    "NLTERateEquationSolver",
+]
+
+
+class NLTERateEquationSolver(ProcessingPlasmaProperty):
+    outputs = ("ion_number_density_nlte", "electron_densities_nlte")
+
+    def calculate(
+        self,
+        gamma,
+        alpha_sp,
+        alpha_stim,
+        coll_ion_coeff,
+        coll_recomb_coeff,
+        partition_function,
+        levels,
+        level_boltzmann_factor,
+        phi,
+        rate_matrix_index,
+        number_density,
+    ):
+        """Calculates ion number densities and electron densities using NLTE ionization.
+
+        Parameters
+        ----------
+        gamma : DataFrame
```

Comment:
Would appreciate short descriptions of these variables

Reply:
Done!